ostd/Cargo.toml

```diff
@@ -42,6 +42,8 @@ unwinding = { version = "0.2.3", default-features = false, features = ["fde-gnu-
 volatile = { version = "0.4.5", features = ["unstable"] }
 xarray = { git = "https://github.com/asterinas/xarray", version = "0.1.0" }
 
+minicov = { git = "https://github.com/YanWQ-monad/minicov", rev = "dcfafdc", version = "0.3", optional = true }
```

Comment:
The small patch can be viewed in https://github.com/YanWQ-monad/minicov/commit/dcfafdca52f3abcb21a10bbe8e03ae2e57e11697. It does two things:
1. adds `-fPIC` flag to compile its C source, so it will use relative address instead of absolute address (the latter will fail on linking),
2. forces to compile the "Linux" platform, because it originally compiled the "Other" platform, which did not match that of the LLVM generated instrumentations.

Reply:
Thanks for the explanation. Do you think these changes may be accepted by the upstream?

Both OSDK and OSTD are published on crates.io. In theory, they cannot use any unpublished dependencies: using a Git repository is not allowed. I expected Cargo to complain about the unpublished dependency in the CI test of "Publish OSDK and OSTD".

```bash
./tools/github_workflows/publish_osdk_and_ostd.sh --dry-run
```

But to my surprise, the CI test passed... I suspect the reason is that the `default` feature does not depend on the `coverage` feature, thus not `minicov`. This makes me wonder what if an OSDK-based project opts in the `coverage` feature manually by editing the `Cargo.toml`.

```toml
[dependencies]
ostd = { version = "0.11.0", features = ["coverage"] }
```

My guess is that `cargo osdk build` would report an error. This means that OSDK's new coverage feature will only be available if the project specifies OSTD as a dependency using a URL or a local FS path (as Asterinas Kernel does).

When writing OSDK's documentation for the coverage feature, please include the following note.

----

**Note.** The code coverage feature of OSDK requires a non-default OSTD feature called `coverage`, which relies on dependencies that are not published on crates.io. To utilize this feature, projects must specify OSTD as a dependency using a Git repository or a local filesystem path in their `Cargo.toml`. For example:

```toml
[dependencies]
ostd = { git = "https://github.com/asterinas/asterinas", rev = "v0.11.0", features = ["coverage"] }
```

Reply:
By the way, all forked dependencies for Asterinas are maintained under the Asterinas Github organization. So I forked `minicov` under the Asterinas organization. Could you submit your changes to [that repository](https://github.com/asterinas/minicov)?